tutorials/topi/intro_topi.py

```diff
@@ -0,0 +1,108 @@
+
+This is an introductory tutorial to TVM Operator Inventory (TOPI).
+
+TOPI provides numpy-style generic operations and schedules with higher abstractions than TVM.
+
+In this tutorial, we will see how TOPI can save from writing boilerplates code in TVM.
+"""
+from __future__ import absolute_import, print_function
+
+import tvm
+import topi
+
+######################################################################
+# Basic example
+# -------------
+# Let's revisit the sum of rows operation (equivalent to :code:`B = numpy.sum(A, axis=1)`') \
+# from :ref:`reduction` tutorial. To compute the sum of rows of a two dimensional TVM tensor A, we should
+# specify the symbolic operation as well as schedule as follows
+#
+n = tvm.var("n")
+m = tvm.var("m")
+A = tvm.placeholder((n, m), name='A')
+k = tvm.reduce_axis((0, m), "k")
+B = tvm.compute((n,), lambda i: tvm.sum(A[i, k], axis=k), name="B")
+s = tvm.create_schedule(B.op)
+
+# and to examine the IR code in human readable format, we can do
+#
+print(tvm.lower(s, [A, B], simple_mode=True))
+
+# However, for such a common operation we had to define the reduce axis ourselves as well as explicit computation with
+# :code: `tvm.compute`. Imagine for more complicated operations how much details we need to provide.
+# Fortunately, we can replace those two lines with simple :code:`topi.sum` much like :code`numpy.sum`
+#
+C = topi.sum(A, axis=1)
+ts = tvm.create_schedule(C.op)
+print(tvm.lower(s, [A, C], simple_mode=True))
+
+######################################################################
+# Generic schedules
+# -----------------
+# Up to now, we have seen an example of how TOPI can save us from writing explicit computations in lower level API.
+# But that doesn't stop here. Still scheduling the operation(s) is done as before. TOPI also provides higher level
+# scheduling recipes depending on a given context. For example, for CUDA, if we want to add `1.0` to `C`,
+#  we can schedule it as injective operation
+#
+D = C + 1.0
+with tvm.target.cuda():
+    sd = topi.generic.schedule_injective(D)
+    print(tvm.lower(sd, [A, D], simple_mode=True))
+
+# And for an element-wise sum with a new tensor `E`, we can do
+#
+E = tvm.placeholder((n,), name="E")
+F = topi.elemwise_sum([D, E], num_args=n)
+with tvm.target.cuda():
+    sf = topi.generic.schedule_elemwise(F)
+    print(tvm.lower(sd, [A, E], simple_mode=True))
+
+# As you can see, scheduled stages of computation accumulate and we can examine them by
+#
+print(sf.stages)
+
+######################################################################
+# Combining schedules
+# -----------------
+# We can continue adding more schedules to operations to generate optimized kernel code. For example,
+#
+a = tvm.placeholder((n,), name="a")
+b = a - 1.0
+c = topi.exp(b)
+d = topi.sum(c)
+ret = c / d
+with tvm.target.create("cuda"):
+    sch = topi.generic.schedule_injective(b)
+    sch = topi.generic.schedule_elemwise(c)
```

Comment:
usually, we don't create multiple schedules like this, but instead build a topi block and call a single schedule function to schedule that